crates/engine/tree/src/tree/payload_processor/sparse_trie.rs

```diff
@@ -129,19 +132,24 @@ where
                 "Updating sparse trie"
             );
 
-            let elapsed = update_sparse_trie(&mut self.trie, update).map_err(|e| {
-                ParallelStateRootError::Other(format!("could not calculate state root: {e:?}"))
-            })?;
+            let elapsed =
+                update_sparse_trie(&mut self.trie, update, &self.blinded_provider_factory)
+                    .map_err(|e| {
+                        ParallelStateRootError::Other(format!(
```

Comment:
idea for follow up:

make other a `Box<dyn Error>` and add `ParallelStateRootError::msg(err: Display)`